drivers/sx127x/sx127x.c

```diff
@@ -207,35 +219,81 @@ static void sx127x_on_dio3_isr(void *arg)
     sx127x_on_dio_isr((sx127x_t*) arg, SX127X_IRQ_DIO3);
 }
 
+#ifdef SX127X_USE_DIO_MULTI
+static void sx127x_on_dio_multi_isr(void *arg)
+{
+    sx127x_on_dio_isr((sx127x_t*) arg, SX127X_IRQ_DIO_MULTI);
+}
+#endif
+
 /* Internal event handlers */
 static int _init_gpios(sx127x_t *dev)
 {
-    int res = gpio_init_int(dev->params.dio0_pin, GPIO_IN, GPIO_RISING,
-                            sx127x_on_dio0_isr, dev);
-    if (res < 0) {
-        DEBUG("[sx127x] error: failed to initialize DIO0 pin\n");
-        return res;
+    int res;
+
+    /* Check if DIO0 pin is defined */
+    if (dev->params.dio0_pin != GPIO_UNDEF) {
+        res = gpio_init_int(dev->params.dio0_pin, GPIO_IN, GPIO_RISING,
+                                sx127x_on_dio0_isr, dev);
+        if (res < 0) {
+            DEBUG("[sx127x] error: failed to initialize DIO0 pin\n");
+            return res;
+        }
+    }
+    else {
+        DEBUG("[sx127x] error: no DIO0 pin defined\n");
```

Comment:
is it still an error ?

Reply:
I would do this differently:
```c
#ifndef SX127X_USE_DIO_MULTI
   /* Check if DIO0 pin is defined */
    if (dev->params.dio0_pin != GPIO_UNDEF) {
        res = gpio_init_int(dev->params.dio0_pin, GPIO_IN, GPIO_RISING,
                            sx127x_on_dio0_isr, dev);
        if (res < 0) {
            DEBUG("[sx127x] error: failed to initialize DIO0 pin\n");
            return res;
        }
    }

    /* check if DIO1 pin is defined */
    if (dev->params.dio1_pin != GPIO_UNDEF) {
        res = gpio_init_int(dev->params.dio1_pin, GPIO_IN, GPIO_RISING,
                            sx127x_on_dio1_isr, dev);
        if (res < 0) {
            DEBUG("[sx127x] error: failed to initialize DIO1 pin\n");
            return res;
        }
    }
#else
/* your new multi dio stuff here */
#endif
```

Reply:
I did it that way because with this radio I think you at least need the DIO0 defined. So in the case that DIO0 is not defined and you are not using the "multi" mode it should fail. If you are using the "multi" mode and no DIO0 is defined, it would check if the MULTI_DIO pin is defined.

Reply:
@aabadie Did you have a chance to look at this?

Reply:
> Did you have a chance to look at this?

Yes, and I still want this to be changed since it can be done simpler. I edited my code snippet above with what I have in mind.
I also think the `sx127x_on_dioX_isr` static functions could be excluded if `SX127X_USE_DIO_MULTI` is defined. Can you check if that save a few bytes ?

Reply:
Modified so it would only initialize the Multi dio pin if `SX127X_USE_DIO_MULTI` is defined. Also excluded the `sx127x_on_diox_isr` functions.